src/Core/Types.hpp

```diff
@@ -35,6 +36,7 @@ using Vector3i = Eigen::Vector3i;
 using Vector4i = Eigen::Vector4i;
 
 using VectorNui = Eigen::Matrix<uint, Eigen::Dynamic, 1>;
+using Vector1ui = Eigen::Matrix<uint, 1, 1>;
```

Comment:
I'm not found of this

Reply:
and so you propose ?
Eigen vector allows compile time introspection, so we can have the same code for all indexed geometry, without a special case for indexed point cloud.

Reply:
well... maybe not defining `Vector1ui` globally but just when called for a PointCloud?